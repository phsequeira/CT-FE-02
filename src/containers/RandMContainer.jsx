import React, { Component } from 'react';
import CharacterList from '../components/characters/CharacterList';
import { findCharacters } from '../services/RandMApi';

export default class RandMContainer extends Component {
    state = {
        characters: []
    }

    async componentDidMount() {
        const characters = await findCharacters();
        this.setState({ characters })
    }

    render() {
        return <CharacterList characters={this.state.characters} />
    }
}