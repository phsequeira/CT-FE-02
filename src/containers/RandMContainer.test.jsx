import React from 'react';
import { render, screen } from '@testing-library/react';
import RandMContainer from './RandMContainer';

describe('RandMContainer', () => {
  it('renders a list of characters to the page', async () => {
    render(<RandMContainer />);

    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).toMatchSnapshot();
  });
});