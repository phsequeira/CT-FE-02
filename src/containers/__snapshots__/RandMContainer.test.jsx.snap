// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandMContainer renders a list of characters to the page 1`] = `
<ul
  aria-label="characters"
/>
`;
